src/frontend/Semantic_check.ml

```diff
@@ -1067,12 +1068,16 @@ let semantic_check_sampling_distribution ~loc id arguments =
   Validate.(
     if
       is_reat_rt_for_suffix "_lpdf"
+      || is_reat_rt_for_suffix "_lupdf"
       || is_reat_rt_for_suffix "_lpmf"
+      || is_reat_rt_for_suffix "_lupmf"
       || is_reat_rt_for_suffix "_log"
```

Comment:
Yeah, will rename to something more appropriate.

Reply:
I am generally not great with naming. Let me know if this might better?

```ocaml
let is_name_w_suffix_sampling_dist suffix =
    Stan_math_signatures.stan_math_returntype (name ^ suffix) argumenttypes
    |> Option.value_map ~default:false ~f:is_real_rt in
  let is_sampling_dist_in_math =
    List.exists ~f:is_name_w_suffix_sampling_dist (Utils.distribution_suffices @ Utils.unnormalized_suffices)
  in
  let is_name_w_suffix_udf_sampling_dist suffix =
    match Symbol_table.look vm (name ^ suffix) with
    | Some (Functions, UFun (listedtypes, ReturnType UReal)) ->
        UnsizedType.check_compatible_arguments_mod_conv name listedtypes
          argumenttypes
    | _ -> false
  in
  let is_udf_sampling_dist =
    List.exists ~f:is_name_w_suffix_udf_sampling_dist (Utils.distribution_suffices @ Utils.unnormalized_suffices)
  in
  Validate.(
    if
      is_sampling_dist_in_math && name <> "binomial_coefficient" && name <> "multiply"
      || is_udf_sampling_dist
    then ok ()
    else error @@ Semantic_error.invalid_sampling_no_such_dist loc name)
```

Reply:
I'd move `<> "binomial_coefficient"` and `<> "multiply"` inside `is_sampling_dist_in_math`.